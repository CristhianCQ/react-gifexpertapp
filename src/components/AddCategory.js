import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [InputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        //  console.log(e.target.value);
          
         setInputValue(e.target.value);
        //  console.log('handleInputChange llamado');
    }

    const handleSubmit = (e) => {
         e.preventDefault();  

         if (InputValue.trim().length > 2) {
            setCategories(cats=> [ InputValue,...cats]);
             // props.setCategories(cats=> [...cats, InputValue]);
            setInputValue('') ;
         }

        
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ InputValue}</p>
            <input
                type="text"
                value={InputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired  
}