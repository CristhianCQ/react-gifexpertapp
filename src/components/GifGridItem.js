import React from 'react'
import PropTypes from 'prop-types'; //impt

export const GifGridItem = ({id, title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}



/* 
    1. enzyme https://enzymejs.github.io/enzyme/ npm i --save-dev enzyme enzyme-adapter-react-16
    2. Enzyme to Jdon https://www.npmjs.com/package/enzyme-to-json npm install --save-dev enzyme-to-json
    3. debe mostrar el componente correctamente
        * shallow
        * wrapper
        * wrapper .toMatchSnap
*/