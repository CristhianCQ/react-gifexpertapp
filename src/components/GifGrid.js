 import React from 'react'
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
 import { GifGridItem } from './GifGridItem';
import PropsTypes from 'prop-types'; //impt

export const GifGrid = ({category}) => {


    const {data:images, loading} = useFetchGifs(category);
 
    // const [images, setImages] = useState([]);


    // useEffect( () => {
    //     getGifs(category).then(setImages);  
    // },[category]);
 
     

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {/* {loading? <p> Loading ...</p>: null    } */}
            {loading && <p className="animate__animated animate__flash"> Loading ...</p> }
            <div className="card-grid">
                
                {
                    images.map((img) => (
                        <GifGridItem 
                            key={img.id}
                            {... img}
                        />
                    ))
                }
                
            </div>
        </>
    )
}

GifGrid.propsTypes = {
    category: PropsTypes.string.isRequired
}