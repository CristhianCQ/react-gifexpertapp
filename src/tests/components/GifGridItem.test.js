 import React from 'react';// imr
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Pruebas';
    const url = 'https://localhost:3000';
    const wrapper =shallow(<GifGridItem  url={url} title={title}/>);
    
    test('should mount correctly the element ', () => {
        expect(wrapper).toMatchSnapshot();    
    })

    test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p'); 

        expect(p.text().trim()).toBe(title);
    })

    test('debe tener el titulo en al atributo alt y un src', () => {
        const img = wrapper.find('img');    

        // console.log(img.html());
        // console.log(img.props().src);

        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
                
    })

    test('debe de tener anime__fadeIn',()=> {
        const div =  wrapper.find('div');
        
                // const numberOfMatches = "animate__fadeIn".filter(x => div.props().className.indexOf(x)> -1).length;
                // expect(numberOfMatches).toBeGreaterThan(0);
                // console.log(numberOfMatches);
        // expect(div.props().className).toMatch(/(animate__fadeIn)/i)

        const className = div.props().className;
        expect(className.includes('animate__fadeIn')).toBe(true);

        
        
    })
    
})