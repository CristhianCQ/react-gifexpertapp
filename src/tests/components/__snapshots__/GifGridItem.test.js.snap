// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should mount correctly the element  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Pruebas"
    src="https://localhost:3000"
  />
  <p>
    Pruebas
  </p>
</div>
`;
